content/en/apps/guides/messaging/rapidpro.md

```diff
@@ -0,0 +1,114 @@
+
+- set the body to relay the message to CHT in the expected format: 
+```json
+@(json(object(
+  "id", run.uuid,
+  "from", replace(urns.tel,"tel:+", "+"),
+  "content", input.text
+)))
+```
+
+![flow_webhook_body](flow_webhook_body.png)
+
+
+### Create a new trigger
+Create a trigger (`/trigger/`) to start the new flow when a message is not handled elsewhere. 
+
+![trigger_select](trigger_select.png)
+
+For more details about RapidPro configuration, please consult the [RapidPro integration documentation]({{% ref "apps/guides/integrations/rapidpro" %}}). 
+
+## CHT Core configuration
+
+### API keys
+
+The API keys should be treated as securely as a password as anyone with access to them will be able to send messages using your account. Therefore, we store them in the CouchDB configuration.
+
+To add the credentials to the admin config you need to either [PUT the value using curl](https://docs.couchdb.org/en/stable/api/server/configuration.html#put--_node-node-name-_config-section-key) or similar:
+
+```sh
+curl -X PUT https://<user>:<pass>@<domain>/_node/couchdb@127.0.0.1/_config/medic-credentials/rapidpro:incoming -d '"<cht_api_key>"'
+curl -X PUT https://<user>:<pass>@<domain>/_node/couchdb@127.0.0.1/_config/medic-credentials/rapidpro:outgoing -d '"<rapidpro_api_key>"'
+```
+
+{{% alert title="Note" %}}
+`couchdb@127.0.0.1` is the local node name, and may be different for you depending on your setup.
+{{% /alert %}}
+
+You can also add it via Fauxton:
+- Navigate to [the Config screen](http://localhost:5984/_utils/#/_config)
+- Click `Add Option`
+- The `Section` should be `medic-credentials`, the `Name` should be `rapidpro:incoming` or `rapidpro:outgoing`, and the value should be the relevant API key.
+- Click `Create`
+- You should then be able to see your credential in the list of configuration shown
+
+### App settings
+
+Update your app settings as follows.
+
+```json
+{
+  "sms": {
+    "outgoing_service": "rapidpro",
+    "rapidpro": {
+      "url": "<RapidPro instance url>"
+    }
+  }
+}
+```
+
+{{% alert title="Rate limiting" %}}
+The RapidPro API endpoints, which this integration relies on, are rate-limited. The number of messages that can be sent and the number of status updates that can be requested is limited. Check [RapidPro API reference](https://rapidpro.io/api/v2/) for more details.
+
+If an action (sending a message or retrieving a status update) fails, it will be retried again later.
+{{% /alert %}}
+
+{{% alert title="Caveat when switching outgoing message services" %}}
+When switching from another outgoing message service to RapidPro, or switching between RapidPro accounts or hosts, make sure that all of the outgoing messages on your instance, that have been relayed through the previous service, are in a final state ("delivered", "failed", "denied", "cleared"). CHT Core will request state updates for non-final messages from RapidPro, which will fail on every retry and consume your request quota.
+{{% /alert %}}
```

Comment:
I had a bit of trouble wrapping my head around these notes. Given that they seem related I've attempted to rewrite it based on my understanding, and combined to a single note:

```suggestion
{{% alert title="Rate limiting" %}}
The RapidPro API endpoints are rate-limited, meaning that requests to send or check the status of messages beyond a certain number per hour will be blocked. The limit is currently 2500 actions per hour, and may change without notice. Check out the [RapidPro API reference](https://rapidpro.io/api/v2/#rate-limiting) for more details. If sending a message or retrieving a status update fails, it will be retried automatically again later.

When the outgoing message service is set to RapidPro or the host/account are changed, CHT Core will request state updates for all messages that aren't in one of the final states: `delivered`, `failed`, `denied`, or `cleared`. Each of these request counts towards your quota. Since the messages are unlikely to exist on the new RapidPro service, these requests will fail on every retry and consume your request quota. It is therefore important that all outgoing messages have a final state before switching RapidPro accounts or hosts. The status can be set to `failed` for messages that should not be resent without user intervention, or to `scheduled` for those that should be automatically sent with the new RapidPro account.
{{% /alert %}}
```

It wasn't clear to me _how_ you'd clear or reset the status of non-final messages, so it would be nice to fill that in, eg "The status can be set {using a script} to `failed`...".

Reply:
Thanks for the feedback @abbyad ! Do you think we should refer to a script that doesn't exist? Should we provide this script? When discussing with @garethbowen , our conclusion was that we could provide a script when the need arises.

Reply:
Sounds good. Pinging @kennsippell @derickl @michaelkohn here as a heads up since app developers may encounter this situation and not know that a script could be built for it, or how to go about doing it efficiently.